### PR TITLE
Add io.bit3.WhatsAppQT

### PR DESCRIPTION
**Pull request by tristanlins**
WhatsAppQT is a qt based desktop app for [WhatsApp Web](https://web.whatsapp.com/).

### CODE REVIEW

**Comment by tristanlins** [comment]
bot, build io.bit3.WhatsAppQT

**Comment by tristanlins** [comment]
bot, build io.bit3.WhatsAppQT

File: io.bit3.WhatsAppQT.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+sdk: org.kde.Sdk
+base: io.qt.qtwebengine.BaseApp
+base-version: '5.14'
+command: WhatsAppQT
+finish-args:
+- "--share=ipc"
+- "--socket=x11"
+- "--socket=wayland"
+- "--share=network"
+- "--filesystem=home"
```
Comment (barthalion) [inline io.bit3.WhatsAppQT.yml:13]: Does it really need full home access?

File: io.bit3.WhatsAppQT.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+app-id: io.bit3.WhatsAppQT
+runtime: org.kde.Platform
+runtime-version: '5.14'
+sdk: org.kde.Sdk
+base: io.qt.qtwebengine.BaseApp
+base-version: '5.14'
+command: WhatsAppQT
+finish-args:
+- "--share=ipc"
+- "--socket=x11"
```
Comment (barthalion) [inline io.bit3.WhatsAppQT.yml:10]: Should be `fallback-x11`, assuming it actually works natively on Wayland.

File: io.bit3.WhatsAppQT.yml (hunk 1 above)
Comment (tristanlins) [reply io.bit3.WhatsAppQT.yml:13]: I almost said it was personal preference. I don't like it when I want to send files and have to move them to Downloads first. And vice versa, if I want to download files, I want to save them directly to a certain location and not in Downloads to move them from there.

But I can switch that to "xdg-downloads" if that's preferred. I can still overwrite the access locally.

File: io.bit3.WhatsAppQT.yml (hunk 2 above)
Comment (tristanlins) [reply io.bit3.WhatsAppQT.yml:10]: I will change it and try it out.

**Comment by tristanlins** [comment]
bot, build io.bit3.WhatsAppQT

**Comment by tristanlins** [comment]
@barthalion I have made the desired changes :-)

File: io.bit3.WhatsAppQT.yml [hunk 3]
```diff
@@ -7,8 +7,7 @@ base-version: '5.14'
 command: WhatsAppQT
 finish-args:
 - "--share=ipc"
-- "--socket=x11"
-- "--socket=wayland"
+- "--socket=fallback-x11"
```
Comment (barthalion) [inline io.bit3.WhatsAppQT.yml:10]: Just to clarify, you still need `--socket=wayland` here, same for ThreemaQt.

Comment (tristanlins) [reply io.bit3.WhatsAppQT.yml:10]: Oops, I read the [documentation](https://docs.flatpak.org/en/latest/flatpak-command-reference.html?highlight=socket#) and thought at first that the Wayland socket would then be passed through. But the second time I read this I realized that I just misunderstood it. 🙈

**Comment by tristanlins** [comment]
bot, build io.bit3.WhatsAppQT

**Comment by barthalion** [comment]
The way Qt handles tray icons is broken:

```
failed to register service "org.kde.StatusNotifierItem-2-1"
```

You should add `--own-name=org.kde.*` to `finish-args`. It's a huge hole but there's no other way currently. With that fixed, LGTM.

**Comment by tristanlins** [comment]
bot, build io.bit3.WhatsAppQT

**Comment by tristanlins** [comment]
> The way Qt handles tray icons is broken:
> 
> ```
> failed to register service "org.kde.StatusNotifierItem-2-1"
> ```

Crazy, I haven't seen any such error message. But maybe there are also differences between `flatpak install` and` flatpak-builder --install`? I've added `--own-name=org.kde.*`.

**Comment by barthalion** [comment]
You probably would if you tried to run TreemaQt and WhatsAppQt at the same time. Qt/KDE uses PID of the process to build service name used to register the icon and it's usually `2` in Flatpak sandbox, causing things to explode.

/merge

**Comment by tristanlins** [comment]
@barthalion Ah, thx for explanation. 
I actually had double tray icons before. But have not seen a suitable message. Which was probably because I never started both clients at the same time via the terminal 🙊

**Pull request closed without merging**